### PR TITLE
Fixed few loggers with mismatching placeholder-parameter count

### PR DESCRIPTION
**Pull request by kssimova**
See openhab/static-code-analysis#213
and openhab/static-code-analysis#237
Signed-off-by: Kristina S. Simova <kssimovaa@gmail.com>

### CODE REVIEW

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 1]
```diff
@@ -433,7 +433,7 @@ protected String formatUndefined(String formatPattern) {
         try {
             return String.format(undefinedFormatPattern, "-");
         } catch (Exception e) {
-            logger.warn("Exception while formatting undefined value [sourcePattern={}, targetPattern={}, {}]",
+            logger.warn("Exception while formatting undefined value [sourcePattern={}, targetPattern={}, exception= ",
```
Comment (htreu) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: Now the format is broken: The closing bracket is missing and also the string placeholder for the exception message is missing. Add it again and provide a `e.getMessage()` instead of the full exception.

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/ZoneTemperatureControlHandler.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.climate.jsonResponseContainer.impl.TemperatureControlStatus;
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.listener.TemperatureControlStatusListener;
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.StructureManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.impl.TemperatureControlManager;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/ZoneTemperatureControlHandler.java]: how is this import related to the fixed log line?

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java [hunk 3]
```diff
@@ -24,13 +24,16 @@
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
 import org.eclipse.smarthome.binding.digitalstrom.handler.BridgeHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.DeviceHandler;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java]: unrelated, please remove.

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java [hunk 4]
```diff
@@ -24,13 +24,16 @@
...
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.Circuit;
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.Device;
 import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.GeneralDeviceInformation;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.constants.OutputModeEnum;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java]: unrelated, please remove.

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java [hunk 5]
```diff
@@ -24,13 +24,16 @@
...
 import org.eclipse.smarthome.binding.digitalstrom.internal.providers.DsDeviceThingTypeProvider;
 import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
 import org.eclipse.smarthome.config.discovery.DiscoveryResult;
 import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.Thing;
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/DeviceDiscoveryService.java]: unrelated, please remove.

File: extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java [hunk 6]
```diff
@@ -61,7 +61,7 @@ protected void openConnection() {
                     try {
                         socket = new Socket(receiverNode.getAddressString(), receiverNode.getPort());
                     } catch (IOException e) {
-                        logger.debug("Could not connect to {} in {}: {}", receiverNode, this.thing.getUID(), e);
+                        logger.debug("Could not connect to {} in {}: ", receiverNode, this.thing.getUID(), e);
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java]: since this is debug logging provide an `e.getMessage()` instead and re-add the placeholder.

File: extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java [hunk 7]
```diff
@@ -88,7 +88,7 @@ protected void closeConnection() {
                 try {
                     socket.close();
                 } catch (IOException e) {
-                    logger.warn("Could not close socket {} in {}: {}", receiverNode, this.thing.getUID(), e);
+                    logger.warn("Could not close socket {} in {}: ", receiverNode, this.thing.getUID(), e);
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java]: since this is warn logging provide an `e.getMessage()` instead and re-add the placeholder.

File: extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java [hunk 8]
```diff
@@ -106,7 +106,7 @@ protected void sendDmxData() {
                     try {
                         socket.getOutputStream().write(universe.getBuffer());
                     } catch (IOException e) {
-                        logger.debug("Could not send to {} in {}: {}", receiverNode, this.thing.getUID(), e);
+                        logger.debug("Could not send to {} in {}: ", receiverNode, this.thing.getUID(), e);
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/Lib485BridgeHandler.java]: since this is debug logging provide an `e.getMessage()` instead and re-add the placeholder.

File: extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/dmxoverethernet/DmxOverEthernetHandler.java [hunk 9]
```diff
@@ -108,7 +108,7 @@ protected void sendDmxData() {
                     try {
                         socket.send(sendPacket);
                     } catch (IOException e) {
-                        logger.debug("Could not send to {} in {}: {}", receiverNode, this.thing.getUID(), e);
+                        logger.debug("Could not send to {} in {}: ", receiverNode, this.thing.getUID(), e);
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/dmxoverethernet/DmxOverEthernetHandler.java]: since this is debug logging provide an `e.getMessage()` instead and re-add the placeholder.

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/ZoneTemperatureControlHandler.java (hunk 2 above)
Comment (kssimova) [reply extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/ZoneTemperatureControlHandler.java]: You are absolutely right, it is not! My "Organize imports"option in Save Actions in Eclipse was selected and with saving changes it must have added these imports.. I did not consider it! Thank you!

**Comment by kaikreuzer** [comment]
@kssimova May I ask you to rebase this PR as it by now has a merge conflict? Thanks!

**Comment by kaikreuzer** [comment]
Thanks!
